vignettes/how-to/box-lsp.Rmd

```diff
@@ -0,0 +1,23 @@
+---
+title: "How-to: Auto-complete in VSCode"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{How-to: Auto-complete in VSCode}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+_Box-module auto-complete only works if one uses VSCode or Vim. It does not have any known adverse effects on RStudio Desktop or Posit Workbench._
+
+# Introduction
+
+Rhino utilizes `{box}` modules to manage large code bases. A disadvantage of `{box}` modules is the lack of syntax auto-complete support in RStudio or VSCode. In VSCode, auto-complete is provided by `{languageserver}` which allows for external parsers. [`{box.lsp}`](https://appsilon.github.io/box.lsp/index.html) is an extension to `{languageserver}` to provide function name and function argument auto-complete support for `{box}` modules. Please refer to the `{box.lsp}` [documentation](https://appsilon.github.io/box.lsp/index.html) for its current capabilities.
+
+# Steps
+
+1. Install `languageserver` to project dependencies: `rhino::pkg_install("languageserver")`.
```

Comment:
`languageserver` should be installed with `renv::install`, not `rhino::pkg_install` - we don't want to add it to `dependencies.R` and `renv.lock`, as it is just a dev dependency on the developer level. Please change it here and add explanation.

Reply:
When we have the ability have a developer mode, we should also move `box.lsp`, `treesitter`, `treesitter.r` out of `Import:`, `renv.lock`, and `dependencies.R`

Reply:
This is not true - `renv::snapshot()` adds only packages listed in `dependencies.R`. That is why using `renv::install` instead of `rhino::pkg_install` makes the installation of `languageserver` local.

Reply:
Right. snapshot checks source files for library and require.